ql/src/java/org/apache/hadoop/hive/ql/metadata/HiveStorageHandler.java

```diff
@@ -613,15 +614,17 @@ default boolean commitInMoveTask() {
    * Commits the inserts for the non-native tables. Used in the {@link org.apache.hadoop.hive.ql.exec.MoveTask}.
    * @param commitProperties Commit properties which are needed for the handler based commit
    * @param operation the operation type
+   * @param suppliedExecutor an ExecutorService to be used by the StorageHandler (optional)
    * @throws HiveException If there is an error during commit
    */
-  default void storageHandlerCommit(Properties commitProperties, Operation operation) throws HiveException {
+  default void storageHandlerCommit(Properties commitProperties, Operation operation, ExecutorService suppliedExecutor)
```

Comment:
1. keep original signature
````
default void storageHandlerCommit(Properties commitProperties, Operation operation) {
  storageHandlerCommit(commitProperties, operation, null)
}
````
2. `executorService`

Reply:
TBH i would rename the method to simply `commit`. why do we need this `storageHandlerCommit` IDK

Reply:
that's not possible I'm afraid, ThreadPools is in iceberg, hive-exec doesn't depend on that

Reply:
that's ok, use `null` then